sklearn/svm/tests/test_svm.py

```diff
@@ -439,6 +439,76 @@ def test_sample_weights():
     assert_array_almost_equal(dual_coef_no_weight, clf.dual_coef_)
 
 
+@pytest.mark.parametrize(
+    "CLF, params",
+    [(svm.SVC,
+      {"err_msg":
+       'Invalid input - all samples have zero/negative weights'}),
+     (svm.NuSVC,
+     {"err_msg":
+      '(negative dimensions are not allowed|nu is infeasible)'}),
+     (svm.SVR,
+     {"err_msg":
+      'Invalid input - all samples have zero/negative weights'}),
+     (svm.NuSVR,
+     {"err_msg":
+      'Invalid input - all samples have zero/negative weights'}),
+     (svm.OneClassSVM,
+     {"err_msg":
+      'Invalid input - all samples have zero/negative weights'})
+     ],
+    ids=['SVC', 'NuSVC', 'SVR', 'NuSVR', 'OneClassSVM']
+)
+@pytest.mark.parametrize(
+    "sample_weight",
+    [[0] * len(Y), [-0.3] * len(Y)],
+    ids=['weights-are-zero', 'weights-are-negative']
+)
+def test_negative_sample_weights_mask_all_samples(CLF, params, sample_weight):
+    clf = CLF(kernel='linear')
+    with pytest.raises(ValueError, match=params['err_msg']):
+        clf.fit(X, Y, sample_weight=sample_weight)
+
+
+@pytest.mark.parametrize(
+    "CLF, params",
```

Comment:
All capital letters are for constant and not class (as a convention). Can you change it for `Classifier` instead

Reply:
done